packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart

```diff
@@ -461,3 +483,68 @@ class DefaultTextEditingShortcuts extends StatelessWidget {
+  'moveUpAndModifySelection:': const ExtendSelectionVerticallyToAdjacentLineIntent(forward: false, collapseSelection: false),
+  'moveDownAndModifySelection:': const ExtendSelectionVerticallyToAdjacentLineIntent(forward: true, collapseSelection: false),
+
+  'moveWordLeft:': const ExtendSelectionToNextWordBoundaryIntent(forward: false, collapseSelection: true),
+  'moveWordRight:': const ExtendSelectionToNextWordBoundaryIntent(forward: true, collapseSelection: true),
+  'moveToBeginningOfParagraph:': const ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: true),
+  'moveToEndOfParagraph:': const ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: true),
+
+  'moveWordLeftAndModifySelection:': const ExtendSelectionToNextWordBoundaryOrCaretLocationIntent(forward: false),
+  'moveWordRightAndModifySelection:': const ExtendSelectionToNextWordBoundaryOrCaretLocationIntent(forward: true),
+  'moveParagraphBackwardAndModifySelection:': const ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: false, collapseAtReversal: true),
+  'moveParagraphForwardAndModifySelection:': const ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: false, collapseAtReversal: true),
+
+  'moveToLeftEndOfLine:': const ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: true),
+  'moveToRightEndOfLine:': const ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: true),
+  'moveToBeginningOfDocument:': const ExtendSelectionToDocumentBoundaryIntent(forward: false, collapseSelection: true),
+  'moveToEndOfDocument:': const ExtendSelectionToDocumentBoundaryIntent(forward: true, collapseSelection: true),
+
+  'moveToLeftEndOfLineAndModifySelection:': const ExpandSelectionToLineBreakIntent(forward: false),
+  'moveToRightEndOfLineAndModifySelection:': const ExpandSelectionToLineBreakIntent(forward: true),
+  'moveToBeginningOfDocumentAndModifySelection:': const ExpandSelectionToDocumentBoundaryIntent(forward: false),
+  'moveToEndOfDocumentAndModifySelection:': const ExpandSelectionToDocumentBoundaryIntent(forward: true),
+
+  'transpose:': const TransposeCharactersIntent(),
+
+  'scrollToBeginningOfDocument:': const ScrollToDocumentBoundaryIntent(forward: false),
+  'scrollToEndOfDocument:': const ScrollToDocumentBoundaryIntent(forward: true),
+
+  // TODO(knopp): Page Up/Down intents are missing (https://github.com/flutter/flutter/pull/105497)
+  'scrollPageUp:': const ScrollToDocumentBoundaryIntent(forward: false),
+  'scrollPageDown:': const ScrollToDocumentBoundaryIntent(forward: true),
+  'pageUpAndModifySelection': const ExpandSelectionToDocumentBoundaryIntent(forward: false),
+  'pageDownAndModifySelection:': const ExpandSelectionToDocumentBoundaryIntent(forward: true),
+
+  // Escape when there's no IME selection
+  'cancelOperation:': const DismissIntent(),
+  // Tab when there's no IME selection
+  'insertTab:': const NextFocusIntent(),
```

Comment:
Because that what would happen without this PR. The `NextFocusIntent` action has own shortcut (`_defaultShortcuts`) and default text editing shortcuts don't override this. With the exception of macO in this PR because we need to feed tab to `TextInputContext`.

Tab handling (and rendering) does need some wrong, but it feels like it is out of scope of this PR.